pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -243,6 +249,42 @@ protected GenericRow doUpdateRecord(GenericRow record, RecordInfo recordInfo) {
     }
   }
 
+  protected GenericRow doOverwriteUpdateRecord(GenericRow record, RecordInfo recordInfo) {
```

Comment:
We can always use this algorithm (loop over each field, only read value when required)

Reply:
that will require some refactoring since this function contains some override-only logic, let me take a look